crates/uv-install-wheel/src/wheel.rs

```diff
@@ -459,32 +461,37 @@ fn install_script(
             use std::fs::Permissions;
             use std::os::unix::fs::PermissionsExt;
 
-            let permissions = fs::metadata(&path)?.permissions();
+            // We fall back to copy when the file is on another drive and if we don't own the file.
+            rename_or_copy.rename_or_copy(&path, &script_absolute)?;
 
-            if permissions.mode() & 0o111 == 0o111 {
-                // If the permissions are already executable, we don't need to change them.
-                rename_with_retry_sync(&path, &script_absolute)?;
-            } else {
-                // If we have to modify the permissions, copy the file, since we might not own it.
-                warn!(
-                    "Copying script from {} to {} (permissions: {:o})",
-                    path.simplified_display(),
-                    script_absolute.simplified_display(),
-                    permissions.mode()
-                );
-
-                uv_fs::copy_atomic_sync(&path, &script_absolute)?;
+            let permissions = fs::metadata(&script_absolute)?.permissions();
 
+            if permissions.mode() & 0o111 != 0o111 {
+                trace!(
+                    "Adjusting permissions for {} from {} to {}",
+                    script_absolute.display(),
+                    permissions.mode(),
+                    permissions.mode() | 0o111
+                );
                 fs::set_permissions(
-                    script_absolute,
+                    &script_absolute,
                     Permissions::from_mode(permissions.mode() | 0o111),
                 )?;
             }
         }
 
         #[cfg(not(unix))]
         {
-            rename_with_retry_sync(&path, &script_absolute)?;
+            // Here, two wrappers over rename are clashing: We want to retry for security software
+            // blocking the file, but we also need the copy fallback is the problem was trying to
+            // move a file cross-drive.
+            match uv_fs::rename_with_retry_sync(&path, &script_absolute) {
+                Ok(()) => (),
+                Err(err) => {
+                    debug!("Failed to rename, falling back to copy: {err}");
+                    fs_err::copy(&path, &script_absolute)?;
+                }
+            }
```

Comment:
:/

Reply:
Should we do `copy_with_retry_sync` here, in the inner loop?

Also, does `rename_with_retry_sync` retry if the error is due to different drives? Hopefully not?

Reply:
Do we have a `copy_with_retry_sync`?

This is mostly guess work, my idea was the security checkers would make the `rename_with_retry_sync loop fail and then the copy would succeed first try, but we can make this more defensive if we have a good way to.

Reply:
Ok, I looked into it myself, and `CrossesDevices` is unstable. Please add a TODO here to catch that error when it stabilizes. Then decide whether you want to add a `copy_with_retry_sync` method (I would suggest doing so, but it's up to you) and go ahead with the merge.

Reply:
Ah now I get what you were getting at.

I now implemented the more defensive version where we retry in both rename or copy. The callback version is kinda ugly but it's only used in this one place. 

I didn't realize there was a dedicated error code for cross drive, i had initially assumed that this is part of some other platform-specific error code. Once it stabilized we can upgrade this code path.

Reply:
If you have suggestions for a prettier version of `with_retry_sync` i'll follow up.